gitlab-runner.yaml

```diff
@@ -0,0 +1,38 @@
+package:
+  name: gitlab-runner
+  version: 16.2.0
+  epoch: 1
+  description: GitLab Runner is the open source project that is used to run your CI/CD jobs and send the results back to GitLab
+  copyright:
+    - license: MIT
```

Comment:
I think `openssl` is a runtime dependency if you need FIPS enabled?

Reply:
You can specify runtime dependencies for this package, or any subpackages, e.g., https://github.com/wolfi-dev/os/blob/be449bc683c36165c23939cb69ec3059d75665dd/gc.yaml#L44

Reply:
The packages would need to be built with `go-fips` to use system OpenSSL, and would need to depend on `libcrypto3` at runtime.

Reply:
@kaniini we set `GOEXPERIMENT=boringcrypto` in our own FIPS builds. Is that necessary if I use your `go-fips` package?

Reply:
Neat! My fork was behind and I hadn't seen the new inline melange pipeline stuff.

I tried building with the `go-fips/build` pipeline and it looks like we import `crypto/boring` where you are patching it  in as `crypto/internal/boring`.

We perform the `boring.Enabled()` check here: https://gitlab.com/gitlab-org/labkit/-/blob/master/fips/fips.go#L32

Reply:
The `go-fips` package implements a wrapper around OpenSSL which allows the use of the OpenSSL FIPS module.  If Gitlab only supports FIPS via BoringSSL, then we can use `GOEXPERIMENT=boringcrypto` instead, but in general our FIPS plan is centered around OpenSSL and its FIPS provider module.

Reply:
@kaniini I don't think we need BoringSSL explicitly, we just referenced the module for runtime checks. I have tried to replicate this behavior by compiling with your `goexperiment.strictfipsruntime` build tag instead of our own `fips` tag.

Does this approach make sense? I wasn't able to get the runtime check to fail (i.e. with or without `libcrypto3` specified as a runtime dependency). How can I validate that it is working?

Reply:
I've removed the FIPS builds for now. We can followup on that in a subsequent PR.